source/auth.c

```diff
@@ -49,6 +58,13 @@ napi_status aws_napi_auth_bind(napi_env env, napi_value exports) {
             .arg_types = {napi_string, napi_string, napi_string},
             .attributes = napi_static,
         },
+        {
+            .name = "newCognito",
+            .method = s_creds_provider_new_cognito,
+            .num_arguments = 4,
+            .arg_types = {napi_undefined, napi_undefined, napi_undefined, napi_undefined},
```

Comment:
Hmmm, we used all the args as `napi_undefined`, but we read some of those as `napi_external` [here](https://github.com/awslabs/aws-crt-nodejs/pull/356/files#diff-ff0326d36d3ec5896c557f1325368efd5a0c9281c31c1969816597d96ea8e806R425). I did a quick search but not seeing anything useful about the difference of the types here... Just pointing out as I feel confusing.

Reply:
undefined means don't enforce any type checking if I'm understanding Colden's original implementation; just following the pattern of earlier bindings here.